r-package/grf/R/input_utilities.R

```diff
@@ -191,19 +196,30 @@ create_data_matrices <- function(X, outcome = NULL, treatment = NULL,
 }
 
 observation_weights <- function(forest) {
-  sample.weights <- if (is.null(forest$sample.weights)) {
-    rep(1, length(forest$Y.orig))
-  } else {
-    forest$sample.weights * length(forest$Y.orig) / sum(forest$sample.weights)
+  # Case 1: No sample.weights
+  if (is.null(forest$sample.weights)) {
+    if (length(forest$clusters) == 0) {
```

Comment:
The condition here should be: `length(forest$clusters) == 0 || !forest$equalize.cluster.weights`